lib/sendgrid/recipient.rb

```diff
@@ -0,0 +1,26 @@
+require 'smtpapi'
+
+module SendGrid
+  class Recipient
+
+    attr_reader :address, :substitutions
+
+    def initialize(address)
+      @address = address
+      @substitutions = {}
+    end
+
+    def add_substitution(key, value)
+      substitutions[key.to_sym] = value
+    end
+
+    def add_to_smtpapi(smtpapi)
+      return if @address.nil? || @substitutions.empty?
```

Comment:
What about raising if this object is initialized with a `nil`? I was weary of doing that since I feel it is one step below full email validation. I opted for the model to simply do nothing here.